scripts/generate_osi_sumo_osg_libs.sh

```diff
@@ -0,0 +1,105 @@
+# in all other cases esmini build process should download the right completely built packages
+#
+# Prerequisites:
+# - look into the single build scripts generate_osg_libs.sh, generate_osi_libs.sh and
+#   generate_sumo_libs.sh
+#
+# Usage:
+# - go into esmini-root-folder/externals
+# - open bash (e.g. Git Bash) in that folder
+# - run the script: ../scripts/generate_osi_sumo_osg_libs.sh
+# - wait (the build process will take approx. 45 minutes depending on...)
+#
+# The folders SUMO/, OpenSceneGrapgh/ and OSI/ will be created as needed for esmini
+#
+#
+# -----------------------------------------------------------------------------------
+# Review and update settings in this section according to your system and preferences
+
+BASE_DIR=`dirname $(realpath $0)`
+echo $BASE_DIR
+OSG_BUILD_DIR=OpenSceneGraph
+OSI_BUILD_DIR=OSI
+SUMO_BUILD_DIR=SUMO
+BUILD_SUB_DIR=manualBuild
+OSG_BUILD_SCRIPT=$BASE_DIR/generate_osg_libs.sh
+OSI_BUILD_SCRIPT=$BASE_DIR/generate_osi_libs.sh
+SUMO_BUILD_SCRIPT=$BASE_DIR/generate_sumo_libs.sh
+
+if [ "$OSTYPE" == "msys" ]; then
+    target_dir="v10"
+    z_exe="/c/Program Files/7-Zip/7z.exe"
+elif [[ "$OSTYPE" == "linux-gnu"* ]]; then
+    target_dir="linux"
+    z_exe=7za
+elif [[ "$OSTYPE" == "darwin"* ]]; then
+    target_dir="mac"
+    z_exe=7z
+else
+    echo Unknown OSTYPE: $OSTYPE
+fi
+
+if [[ ! -d "$OSG_BUILD_DIR/$target_dir" ]]; then
+    if [ ! -d $OSG_BUILD_DIR ]; then
+    	mkdir "$OSG_BUILD_DIR"
+    fi
+    if [[ ! -d "$OSG_BUILD_DIR/$BUILD_SUB_DIR" ]]; then
+    	mkdir "$OSG_BUILD_DIR/$BUILD_SUB_DIR"
+    fi
+    cd "$OSG_BUILD_DIR/$BUILD_SUB_DIR"
+    $OSG_BUILD_SCRIPT
+    cd ..
+    "$z_exe" x "$BUILD_SUB_DIR/osg_$target_dir.7z"
+    if [ "$OSTYPE" == "msys" ]; then
+        rmdir "$BUILD_SUB_DIR" /s /q
+    elif [[ "$OSTYPE" == "darwin"* ]] || [[ "$OSTYPE" == "linux"* ]]; then
+        rm -rf "$BUILD_SUB_DIR"
+    fi
+    cd ..
+fi
+if [[ ! -d "$OSI_BUILD_DIR/$target_dir" ]]; then
+    if [[ ! -d "$OSI_BUILD_DIR" ]]; then
+    	mkdir "$OSI_BUILD_DIR"
+    fi
+    if [[ ! -d "$OSI_BUILD_DIR/$BUILD_SUB_DIR" ]]; then
+    	mkdir "$OSI_BUILD_DIR/$BUILD_SUB_DIR"
+    fi
+    cd "$OSI_BUILD_DIR/$BUILD_SUB_DIR"
+    $OSI_BUILD_SCRIPT
+    cd ..
+    "$z_exe" x "$BUILD_SUB_DIR/osi_$target_dir.7z"
+    if [ "$OSTYPE" == "msys" ]; then
+        rmdir "$BUILD_SUB_DIR" /s /q
+    elif [[ "$OSTYPE" == "darwin"* ]] || [[ "$OSTYPE" == "linux"* ]]; then
+        rm -rf "$BUILD_SUB_DIR"
+    fi
+    cd ..
+fi
+if [[ ! -d "$SUMO_BUILD_DIR/$target_dir" ]]; then
+    if [[ ! -d "$SUMO_BUILD_DIR" ]]; then
+    	mkdir "$SUMO_BUILD_DIR"
+    fi
+    if [[ ! -d "$SUMO_BUILD_DIR/$BUILD_SUB_DIR" ]]; then
+    	mkdir "$SUMO_BUILD_DIR/$BUILD_SUB_DIR"
+    fi
+    cd "$SUMO_BUILD_DIR/$BUILD_SUB_DIR"
+    $SUMO_BUILD_SCRIPT
+    cd ..
+    "$z_exe" x "$BUILD_SUB_DIR/sumo_$target_dir.7z"
+    if [ "$OSTYPE" == "msys" ]; then
+        rmdir "$BUILD_SUB_DIR" /s /q
```

Comment:
This fails in git bash on Win10. What do you think of changing into
`rm -rf "$BUILD_SUB_DIR"`
?
Actually, then we can skip this if-statement altogether. What do you think?

Reply:
This is the difference between Win cmd snd powershell? Thought rm -rf does not work for win cmd, are you using power shell? But for sure if rm -rf works for you, you can delete the if condition, have tested only on linux, rest searched via google

Reply:
I see. I was lazy enough to limit the Windows case to only support (Git) bash. I think msys/msys2 bash would work equally good (guess it's the same). 

I agree that it would be complex to support all possible command interpreters (PowerShell, cmd, bash...). So I think it's perfectly OK to stick with bash support for now. At least we should postpone hard work until it's needed :)

I found and fixed a few flaws in the OSG script on Windows that was introduced by myself in last update. 

Now everything seems to compile perfectly on Windows as well - in one single command! Really nice 👍

Reply:
ok, then please remove the if conditions if not already done

Reply:
I will, and then I think we're ready for merge